### PR TITLE
Playing cards QoL changes

### PR DESCRIPTION
**Pull request by quardbreak**
Stole some qol changes from Infinity Station:
* Added compact type
* Added click on gambling table to deal with cards
* Some misc output messages like deck examine.

### CODE REVIEW

File: code/modules/games/cards.dm [hunk 1]
```diff
@@ -54,7 +54,9 @@
 
 /obj/item/deck/cards/Initialize()
 	. = ..()
+	return INITIALIZE_HINT_LATELOAD
 
+/obj/item/deck/cards/LateInitialize()
```
Comment (afterthought2) [inline code/modules/games/cards.dm]: Why the move to lateload?

File: code/modules/games/cards.dm [hunk 2]
```diff
@@ -78,18 +80,59 @@
...
 		P.name = "joker"
 		P.card_icon = "joker"
 		cards += P
 
+/obj/item/deck/compact
+	name = "compact deck of cards"
+	desc = "A deck of playing cards. Looks like this one hasn't numbers from two to five, and jokers."
+	icon_state = "deck"
+
+/obj/item/deck/compact/LateInitialize()
```
Comment (afterthought2) [inline code/modules/games/cards.dm]: If it's to do a hard override without calling parent, just use a custom proc that you call from Initialize instead.

File: code/modules/games/cards.dm [hunk 3]
```diff
@@ -185,16 +232,20 @@
...
-	if(!Adjacent(usr) || !over.Adjacent(usr)) return // should stop you from dragging through windows
-
-	if(!ishuman(over) || !(over in viewers(3))) return
-
-	if(!cards.len)
-		to_chat(usr, "There are no cards in the deck.")
-		return
-
-	deal_at(usr, over)
+	if(over == usr && !usr.restrained() && !usr.stat && (usr.contents.Find(src) || in_range(src, usr)))
```
Comment (afterthought2) [inline code/modules/games/cards.dm]: `incapacitated` should cover some of these checks.

File: code/modules/tables/tables.dm [hunk 4]
```diff
@@ -130,6 +130,16 @@
...
 		if(H.cards && H.cards.len == 1)
 			usr.visible_message("\The [user] plays \the [H.cards[1].name].")
+
+	if(istype(W, /obj/item/deck)) //playing cards
+		if(user.a_intent == I_GRAB)
+			var/obj/item/deck/D = W
+			if(!D.cards.len)
+				to_chat(usr, "There are no cards in the deck.")
+				return
+			D.deal_at(usr, src)
```
Comment (afterthought2) [inline code/modules/tables/tables.dm]: `user`. Same two lines above.

File: code/modules/tables/tables.dm [hunk 5]
```diff
@@ -130,6 +130,16 @@
 		var/obj/item/hand/H = W
 		if(H.cards && H.cards.len == 1)
 			usr.visible_message("\The [user] plays \the [H.cards[1].name].")
+
+	if(istype(W, /obj/item/deck)) //playing cards
+		if(user.a_intent == I_GRAB)
```
Comment (afterthought2) [inline code/modules/tables/tables.dm:135]: Wait, why?

Comment (quardbreak) [reply code/modules/tables/tables.dm:135]: Every attackby will trigger this. We make it check grab intent so it won't call deal proc.

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**